### PR TITLE
🐞 Fix Rich Progress with Patchcore Training

### PR DESCRIPTION
**Pull request by ashwinvaidya17**
## 📝 Description

- Since coreset subsampling uses `track` from `rich`, it breaks when we use any other progress bar. This caches the `LiveDisplay` before using `track` in subsampling.

## ✨ Changes

Select what type of change your PR is:

- [x] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🔨 Refactor (non-breaking change which refactors the code base)
- [ ] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 Documentation update
- [ ] 🔒 Security update

## ✅ Checklist

Before you submit your pull request, please make sure you have completed the following steps:

- [ ] 📋 I have summarized my changes in the [CHANGELOG](https://github.com/openvinotoolkit/anomalib/blob/main/CHANGELOG.md) and followed the guidelines for my type of change (skip for minor changes, documentation updates, and test enhancements).
- [ ] 📚 I have made the necessary updates to the documentation (if applicable).
- [ ] 🧪 I have written tests that support my changes and prove that my fix is effective or my feature works (if applicable).

For more information about code review checklists, see the [Code Review Checklist](https://github.com/openvinotoolkit/anomalib/blob/main/docs/source/markdown/guides/developer/code_review_checklist.md).

### CODE REVIEW

File: src/anomalib/engine/engine.py [hunk 1]
```diff
@@ -406,7 +406,7 @@ def _setup_transform(
 
     def _setup_anomalib_callbacks(self) -> None:
         """Set up callbacks for the trainer."""
-        _callbacks: list[Callback] = []
+        _callbacks: list[Callback] = [RichProgressBar(), RichModelSummary()]
```
Comment (ashwinvaidya17) [inline src/anomalib/engine/engine.py:409]: This change is not directly related to the bug but I am now using `RichProgressBar` in the trainer. I don't mind if we drop this as it is not necessary

**Comment by ashwinvaidya17** [comment]
@blaz-r I can't reproduce the error with Padim but maybe the fix in this PR might work for you.

**Review by samet-akcay — APPROVED** [review]

**Comment by blaz-r** [comment]
I'll check it out now.

**Comment by blaz-r** [comment]
I tried to run it on ubuntu through wsl and on windows directly. While I didn't manage to get it working on ubuntu, It work on windows now. The only problem is with filename, that is not valid on windows. I left a comment with a suggestion on how it can be easily fixed.

**Comment by ashwinvaidya17** [comment]
What's the error on ubuntu?

**Comment by ashwinvaidya17** [comment]
Alright in that case I'll merge it as is as I am using Ubuntu and I can't reproduce it. We can always revisit this in-case more people face this issue.

**Comment by blaz-r** [comment]
Okay
